File: src/main/java/com/naminhyeok/fantazzk/room/domain/RoomId.java
package com.naminhyeok.fantazzk.room.domain;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.util.UUID;
import org.jmolecules.ddd.types.Identifier;

public record RoomId(UUID roomId) implements Identifier {
    @Converter(autoApply = true)
    static final class JpaConverter implements AttributeConverter<RoomId, UUID> {
        @Override
        public UUID convertToDatabaseColumn(RoomId attribute) {
            return attribute == null ? null : attribute.roomId();
        }

        @Override
        public RoomId convertToEntityAttribute(UUID dbData) {
            return dbData == null ? null : new RoomId(dbData);
        }
    }
}

File: src/main/kotlin/com/naminhyeok/fantazzk/room/domain/RoomId.kt
package com.naminhyeok.fantazzk.room.domain

import org.jmolecules.ddd.types.Identifier

data class RoomId(
    val value: Long,
) : Identifier {
    init {
        require(value > 0L) { "RoomId는 1 이상이어야 합니다" }
    }

    companion object {
        fun from(value: Long): RoomId = RoomId(value)

        fun fromOrNull(value: Long?): RoomId? = value?.takeIf { it != 0L }?.let(::from)
    }
}
